source/jormungandr/jormungandr/scenarios/tests/journey_compare_tests.py

```diff
@@ -519,13 +519,85 @@ def test_not_too_late_journeys():
 
-    journey_filter._filter_not_coherent_journeys(list(journeys_gen(responses)),
+    journey_filter._filter_too_long_journeys(list(journeys_gen(responses)),
                                                  MockInstance(), request)
 
     assert 'to_delete' not in journey1.tags
     assert 'to_delete' not in journey2.tags
 
 
+def test_too_late_journeys_but_better_mode():
+    request = {'datetime': 1000000,
+               'clockwise': True,
+               '_night_bus_filter_max_factor': 2,
+               '_night_bus_filter_base_factor': 9,
+               }
+    responses = [response_pb2.Response()]
+    car = responses[0].journeys.add()
+    car.departure_date_time = 1001042
+    car.arrival_date_time = 1002000
+    car.tags.append('car')
+
+    responses.append(response_pb2.Response())
+    bike = responses[-1].journeys.add()
+    bike.departure_date_time = 1001042
+    bike.arrival_date_time = 1004010 # car * 2 + 10
+    bike.tags.append('bike')
+
+    responses.append(response_pb2.Response())
+    bss = responses[-1].journeys.add()
+    bss.departure_date_time = 1001042
+    bss.arrival_date_time = 1008030 # bike * 2 + 10
+    bss.tags.append('bss')
+
+    responses.append(response_pb2.Response())
+    walking = responses[-1].journeys.add()
+    walking.departure_date_time = 1001042
+    walking.arrival_date_time = 1016070 # bss * 2 + 10
+    walking.tags.append('walking')
+
+    journey_filter._filter_too_long_journeys(list(journeys_gen(responses)),
+                                             MockInstance(), request)
+
+    assert all('to_delete' not in j.tags for r in responses for j in r.journeys)
+
+
+def test_too_late_journeys_and_worst_mode():
+    request = {'datetime': 1000000,
+               'clockwise': True,
+               '_night_bus_filter_max_factor': 2,
+               '_night_bus_filter_base_factor': 9,
+               }
+    responses = [response_pb2.Response()]
+    walking = responses[0].journeys.add()
+    walking.departure_date_time = 1001042
+    walking.arrival_date_time = 1002000
+    walking.tags.append('walking')
+
+    responses.append(response_pb2.Response())
+    bss = responses[-1].journeys.add()
+    bss.departure_date_time = 1001042
+    bss.arrival_date_time = 1004010 # walking * 2 + 10
+    bss.tags.append('bss')
+
+    responses.append(response_pb2.Response())
+    bike = responses[-1].journeys.add()
+    bike.departure_date_time = 1001042
+    bike.arrival_date_time = 1008030 # bss * 2 + 10
+    bike.tags.append('bike')
+
+    responses.append(response_pb2.Response())
+    car = responses[-1].journeys.add()
+    car.departure_date_time = 1001042
+    car.arrival_date_time = 1016070 # bike * 2 + 10
+    car.tags.append('car')
+
+    journey_filter._filter_too_long_journeys(list(journeys_gen(responses)),
+                                             MockInstance(), request)
+
+    assert sum('to_delete' not in j.tags for r in responses for j in r.journeys) == 1
```

Comment:
I'd prefer `['to_delete' not in j.tags for r in responses for j in r.journeys].count(True) == 1` but your code is nice, too